MIPs/mip-x-multichainapi.md

```diff
@@ -0,0 +1,243 @@
+
+### CAIP-27 - Invoke RPC Requests for an Authorization Scope
+A dapp can invoke RPC requests for an authorization scope by making [CAIP-27](https://chainagnostic.org/CAIPs/caip-27) `wallet_invokeMethod` calls.
+
+> **Note:** a `sessionId` parameter is not supported and will be ignored, if included.
+
+### CAIP-312 - Retrieve Authorization Scopes
+A dapp can retrieve a [CAIP-312](https://chainagnostic.org/CAIPs/caip-312) multichain session object by calling `wallet_getSession` to request its authorization scopes at any time.
+
+> **Note:** a `sessionId` parameter is not supported and will be ignored, if included.
+
+### CAIP-311 - Notification of Changes to Authorization Scopes
+A dapp can listen for [CAIP-311](https://chainagnostic.org/CAIPs/caip-311) `wallet_sessionChanged` events in order to get notified when there are changes to its authorization scopes.
+
+> **Note:** A `sessionId` will not be included as part of the event.
+
+### CAIP-285 - Revoke Authorization Scopes
+A dapp can make a [CAIP-285](https://chainagnostic.org/CAIPs/caip-285) `wallet_revokeSession` request to revoke all of its authorization scopes.
+
+> **Note:** a `sessionId` parameter is not supported and will be ignored, if included.
+
+### API Specification Document
+An OpenRPC [specification for the Multichain API](https://github.com/MetaMask/api-specs/blob/main/multichain/openrpc.yaml) can be found in the [api-specs](https://github.com/MetaMask/api-specs) repository.
+
+## Caveats
+The CAIPs referenced in this proposal are still in Draft or Review status with CASA and may be subject to material changes.
+
+### API Access
+As part of the Multichain API, MetaMask is leading the adoption of alternative methods for browser extension wallet communication. Currently, browser extension wallets inject a JavaScript object to communicate with each site. But this approach comes with privacy, security, reliability, and performance drawbacks.
+
+- The MetaMask Browser Extension will adopt an alternative method of communication through the [externally_connectable](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable) feature, which is now supported by most modern web browsers with extension frameworks.
+- When a site is accessed through a browser that does not support [externally_connectable](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable), the Multichain API will be available through an alternative communication method. 
+
+In summary, the following methods will solely be accessible through the new communication methods:
+
+- `wallet_createSession` (CAIP-25)
+- `wallet_sessionChanged` (CAIP-311)
+- `wallet_getSession` (CAIP-312)
+- `wallet_revokeSession` (CAIP-285)
+- `wallet_invokeMethod` (CAIP-27)
+- `wallet_notify` (CAIP-319)
+
+(API Delivery & Wallet Discovery CAIPs to be added in this section)
```

Comment:
Assuming this will be dealt with later

Reply:
Yes. I think we should wait to link once we at least have draft CAIPs posted.